numpy/core/tests/test_datetime.py

```diff
@@ -20,6 +21,18 @@
     _has_pytz = False
 
 
+
+def assert_array_equiv(x, y, err_msg='', verbose=True):
+    """
+    Raises an AssertionError if two array_like objects are not equivalent.
+
+    Equivalent objects are either equal or each NaN/NaT in each position.
+    """
+    assert_array_compare(lambda x, y: (x == y) | ((x != x) & (y != y)),
+                         x, y, err_msg=err_msg, verbose=verbose,
+                         header='Arrays are not equivalent')
+
```

Comment:
Can you show me what "the opposite for NaN" means? I thought it was the same behavior:

```
In [3]: import numpy as np

In [4]: np.testing.assert_array_equal([np.nan], [np.nan])

In [5]: np.testing.assert_array_equal([np.nan], [3])
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
```